From 5d8676b5ff66252a7e61850d05f31ac98d9630de Mon Sep 17 00:00:00 2001
From: Emergi Idan <idan.emergi@intel.com>
Date: Mon, 27 Jan 2020 17:34:08 +0200
Subject: [PATCH] PONRTSYS-6536: b step verification

Fix checker fallback configuration for b-step
Fix parser last field vector debug print

Signed-off-by: Emergi Idan <idan.emergi@intel.com>
---
 drivers/net/ppv4/pp-hal/checker/checker.c       | 2 +-
 drivers/net/ppv4/pp-hal/parser/parser_debugfs.c | 5 ++---
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ppv4/pp-hal/checker/checker.c b/drivers/net/ppv4/pp-hal/checker/checker.c
index 76b0c66335af..992f00476f3e 100644
--- a/drivers/net/ppv4/pp-hal/checker/checker.c
+++ b/drivers/net/ppv4/pp-hal/checker/checker.c
@@ -1087,7 +1087,7 @@ s32 chk_init(struct pp_chk_init_param *init_param)
 	reg = PP_REG_RD32(PP_CHK_STAGE2_MATCH_FALLBACK_REG);
 	reg = PP_FIELD_MOD(PP_CHK_STAGE2_MATCH_FALLBACK_DIS_SW_ERR_ALT_SI_MSK,
 			   1, reg);
-	if (!IS_ENABLED(CONFIG_PPV4_FPGA)) /* no ingress UC in FPGA */
+	if (pp_silicon_step_get() == PP_SSTEP_A)
 		reg = PP_FIELD_MOD(BIT(SI_CHCK_FLAG_DVRT_FRAG), 1, reg);
 	PP_REG_WR32(PP_CHK_STAGE2_MATCH_FALLBACK_REG, reg);
 
diff --git a/drivers/net/ppv4/pp-hal/parser/parser_debugfs.c b/drivers/net/ppv4/pp-hal/parser/parser_debugfs.c
index 06233e71f508..0d9a37dc0878 100644
--- a/drivers/net/ppv4/pp-hal/parser/parser_debugfs.c
+++ b/drivers/net/ppv4/pp-hal/parser/parser_debugfs.c
@@ -2318,9 +2318,8 @@ void __prsr_fv_raw_show(struct seq_file *f)
 	seq_puts(f, " |               |  Unmasked   |  Masked     |\n");
 	seq_puts(f, " +---------------+-------------+-------------+\n");
 	for (i = 0; i < (sizeof(fv) / sizeof(u32)); i++) {
-		seq_printf(f, " |   FV WORD%02u   | %#010x  | %#010x  |\n",
-			   i, be32_to_cpu(((u32 *)&fv)[i]),
-			   be32_to_cpu(((u32 *)&masked_fv)[i]));
+		seq_printf(f, " |   FV WORD%02u   | %#010x  | %#010x  |\n", i,
+			   (((u32 *)&fv)[i]), (((u32 *)&masked_fv)[i]));
 	}
 	seq_puts(f, " +---------------+-------------+-------------+\n");
 }
